### ISSUE TITLE
Output is incorrect for InlineHAL VMVX backend

### ISSUE THREAD

**Issue by lundong** [iree-org/iree#12220]
### What happened?

For some TFLite models, after importing them via TOSA, iree-compile can compile the imported TOSA MLIR into vmfb for the VMVX backend. However, there exists output discrepancy between regular HAL and inline HAL. Regular HAL output result is correct; inline HAL output is incorrect.

### Steps to reproduce your issue

1. Build IREE
2. Download the sample person detection model from: https://storage.googleapis.com/iree-model-artifacts/person_detect.tflite
3. Import the downloaded TFLite model
path/to/iree-import-tflite path/to/person_detect.tflite -o=/tmp/person_detect.mlir
4. Use iree-compile to compile the MLIR
4.1 VMVX with regular HAL
/path/to/iree-compile --iree-input-type=tosa --iree-hal-target-backends=vmvx /tmp/person_detect.mlir -o /tmp/person_detect.vmfb
4.2 VMVX with inline HAL
/path/to/iree-compile --iree-input-type=tosa --iree-hal-target-backends=vmvx-inline --iree-execution-model=inline-static /tmp/person_detect.mlir -o /tmp/person_detect_inline.vmfb
5. Run the model with all zero input and see different results
5.1 iree-run-module with VMVX regular HAL (reference correct output)
path/to/iree-run-module --module=/tmp/person_detect.vmfb --function=main --input="1x96x96x1xi8=0"
EXEC @main
result[0]: hal.buffer_view
1x2xi8=[73 -73]
5.2 iree-run-module with VMVX InlineHAL (incorrect output)
path/to/iree-run-module --module=/tmp/person_detect_inline.vmfb --function=main --input="1x96x96x1xi8=0"
EXEC @main
result[0]: hal.buffer_view
1x2xi8=[-42 42]

### What component(s) does this issue relate to?

Compiler

### Version information

Latest IREE ToT

### Additional context

The issue may have existed for quite a while.

**Comment by lundong** [comment]
FYI @hcindyl @jpienaar

**Comment by jpienaar** [comment]
Could you check via LLVM backend too?

**Comment by lundong** [comment]
This issue doesn’t exist for LLVM backend. VMVX only.

**Comment by benvanik** [comment]
You can try adding `--iree-flow-inject-dispatch-tracing` to the compiler invocations: that should print out all the tensors that move between dispatches to stdout and make it possible via diff to find where things go wrong. Doing that would help narrow down which dispatch may be causing the problems so we can reproduce it.

**Comment by allieculp** [comment]
@lundong Is there any update here?

**Comment by lundong** [comment]
> @lundong Is there any update here?

No. Due to the open issue #12586 (that appeared later than this issue), now Inline HAL compilation actually fails on VMVX backend so can't generate any output. Not sure if this issue will still exist after #12586 is resolved. Will revisit it by then.

**Comment by lundong** [comment]
Update: issue #12586 has been resolved. However, due to another open issue #12778, Inline HAL compilation still fails on VMVX backend for some models, including the aforementioned example. Will revisit this after #12778 is resolved.
